test/conformance/conformance_task_group.cpp

```diff
@@ -0,0 +1,264 @@
+    });
+    CHECK_MESSAGE(run == false, "delayed task should not be run until run(task_handle) is called");
+    tg.run_and_wait(std::move(h));
+    CHECK_MESSAGE(run == true, "Delayed task should be completed when task_group::wait exits");
+
+    CHECK_MESSAGE(h == nullptr, "Delayed task can be executed only once");
+}
+
+//! Test for empty check
+//! \brief \ref interface
+TEST_CASE("Task handle empty check"){
+    oneapi::tbb::task_group tg;
+
+    oneapi::tbb::task_handle h;
+
+    bool empty = (h == nullptr);
+    CHECK_MESSAGE(empty, "default constructed task_handle should be empty");
+
+    h = tg.defer([]{});
+
+    CHECK_MESSAGE(h != nullptr, "delayed task returned by task_group::delayed should not be empty");
+}
+
+//! Test for comparison operations
+//! \brief \ref interface
+TEST_CASE("Task handle comparison/empty checks"){
+    oneapi::tbb::task_group tg;
+
+    oneapi::tbb::task_handle h;
+
+    bool empty =  ! static_cast<bool>(h);
+    CHECK_MESSAGE(empty, "default constructed task_handle should be empty");
+    CHECK_MESSAGE(h == nullptr, "default constructed task_handle should be empty");
+    CHECK_MESSAGE(nullptr == h, "default constructed task_handle should be empty");
+
+    h = tg.defer([]{});
+
+    CHECK_MESSAGE(h != nullptr, "deferred task returned by task_group::defer() should not be empty");
+    CHECK_MESSAGE(nullptr != h, "deferred task returned by task_group::defer() should not be empty");
+
+}
+
+//! Test for task_handle being non copyable
+//! \brief \ref interface
+TEST_CASE("Task handle being non copyable"){
+    static_assert(
+              (!std::is_copy_constructible<oneapi::tbb::task_handle>::value)
+            &&(!std::is_copy_assignable<oneapi::tbb::task_handle>::value),
+            "oneapi::tbb::task_handle should be non copyable");
+}
+//! Test that task_handle prolongs task_group::wait
+//! \brief \ref requirement
+TEST_CASE("Task handle blocks wait"){
+    oneapi::tbb::task_group tg;
+
+    std::atomic<bool> completed  {false};
+    std::atomic<bool> start_wait {false};
+    std::atomic<bool> thread_started{false};
+
+    oneapi::tbb::task_handle h = tg.defer([&]{
+        completed = true;
+    });
+
+    std::thread wait_thread {[&]{
+        CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) is called");
+
+        thread_started = true;
+        utils::SpinWaitUntilEq(start_wait, true);
+        tg.wait();
+        CHECK_MESSAGE(completed == true, "Deferred task should be completed when task_group::wait exits");
+    }};
+
+    utils::SpinWaitUntilEq(thread_started, true);
+    CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) is called");
+
+    tg.run(std::move(h));
+    //TODO: more accurate test (with fixed number of threads (1 ?) to guarantee correctness of following assert)
+    //CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) and wait is called");
```

Comment:
I am not sure that I understand the TODO. Why deferred task should not be run until wait is called?

Reply:
IIRC, the idea was to make a test more stable by making it single-threaded and utilize the fact that task added by `run` is not guaranteed to execute until `wait` is called. 

BTW for some unknown reason [task_group spec](https://spec.oneapi.io/versions/latest/elements/oneTBB/source/task_scheduler/task_group/task_group_cls.html) does not mention this.